deepchecks/vision/checks/performance/single_dataset_scalar_performance.py

```diff
@@ -0,0 +1,148 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing a check for computing a scalar performance metric for a single dataset."""
+import typing as t
+import warnings
+
+import torch
+from ignite.metrics import Accuracy, Metric
+
+from deepchecks.core import CheckResult, ConditionResult, DatasetKind
+from deepchecks.core.condition import ConditionCategory
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.vision import Batch, Context, SingleDatasetCheck
+from deepchecks.vision.metrics_utils.object_detection_precision_recall import \
+    ObjectDetectionAveragePrecision
+from deepchecks.vision.vision_data import TaskType
+
+__all__ = ['SingleDatasetScalarPerformance']
+
+
+class SingleDatasetScalarPerformance(SingleDatasetCheck):
+    """Calculate a performance metric as a scalar for a given model and a given dataset.
+
+    Parameters
+    ----------
+        metric: Metric, default: None
+        An ignite.Metric object whose score should be used. If None is given, use the default metric.
+        reduce: torch function, default: None
+        The function to reduce the scores tensor into a single scalar. For metrics that return a scalar use None
+        (default).
+    """
+
+    def __init__(self,
+                 metric: Metric = None,
+                 reduce: t.Callable = None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.metric = metric
+        self.reduce = reduce
+        if 'metric_name' in kwargs:
+            self.metric_name = kwargs['metric_name']
+
+        if 'reduce_name' in kwargs:
+            self.reduce_name = kwargs['reduce_name']
```

Comment:
it would be better to just have the var as None instead of checking `not hasattr(self, 'reduce_name')`
```suggestion
self.reduce_name = kwargs.get('reduce_name') or (reduce.__name__ if reduce else None)
```

Reply:
same for metric_name

Reply:
better to just add it reduce_name and metric_name to the function parameters with None as default (same for metric_name)

Reply:
why not in function params?

Reply:
Since they only affect the display and not the actual functionality, they seem more like kwargs to me, but I'm open to change

Reply:
how much does it matter if we have the or not? won't name be enough?

Reply:
@noamzbr your call

Reply:
I'm for Check params - we have other checks with params that control displays, and I think it should be obvious to the user how to control that (display is important)

Reply:
Changed to params